xcm/xcm-builder/src/fungibles_adapter.rs

```diff
@@ -38,18 +38,24 @@ impl<Prefix: Get<MultiLocation>, AssetId: Clone, ConvertAssetId: Convert<u128, A
 	fn convert_ref(id: impl Borrow<MultiLocation>) -> result::Result<AssetId, ()> {
 		let prefix = Prefix::get();
 		let id = id.borrow();
-		if !prefix.iter().enumerate().all(|(index, item)| id.at(index) == Some(item)) {
+		if prefix.parent_count() != id.parent_count() ||
+			prefix
+				.interior()
+				.iter()
+				.enumerate()
+				.any(|(index, junction)| id.interior().at(index) != Some(junction))
+		{
```

Comment:
isn't this block basically the definition of `Eq` or `PartialEq` for `MultiLocation`?

Reply:
Huh, yeah, I wonder why `Eq` wasn't used before in this block...?

Reply:
Ah ok, I remember why -- `id` and `prefix` may have different lengths. In case `prefix` is shorter, then we want to make sure that `prefix` is a proper prefix of `id`.